torch/csrc/jit/codegen/cuda/scheduler/normalization.cpp

```diff
@@ -351,6 +351,16 @@ ReductionParams OuterNormalizationHeuristic(
   remainder_in_output = ceilDiv(num_outputs_for_reduction, bdimx);
   remainder_in_reduction = ceilDiv(remainder_in_reduction, bdimy);
 
+  // Try to fill the width of the machine by adjust the
+  // the threads devoted to output vs threads used to calculate
+  // the reduction
+  if (remainder_in_output < device_multiprocessor_count) {
+    // TODO: Use cache sector size / data type size instead?
+    bdimx = lastPow2(
```

Comment:
Does it still work well if you make sure this isn't smaller than `warp_size`?

Reply:
I was not sure what you meant by this comment as I am changing the `warp_size` to be smaller. (ie number of outputs processed by `bdimx`)